sdks/python/gen_protos.py

```diff
@@ -110,14 +112,17 @@ def generate_proto_files(force=False):
             'Protoc returned non-zero status (see logs for details): '
             '%s' % ret_code)
 
-    if sys.version_info[0] >= 3:
-      ret_code = subprocess.call(
-          ["futurize", "--both-stages", "--write", "--verbose", "--no-diff",
-           out_dir])
+    ret_code = subprocess.call(["pip", "install", "future==0.16.0"])
```

Comment:
Future needs to be installed to run the futurize step on the generated proto files. Since gen_protos.py is called from setup.py before building, required packages are not installed yet. I tried using setup_requires in setup.py, but this also did not work.

This step only runs when the proto files are regenerated, so it shouldn't be too big of a problem. However, I'm open to suggestions if there is a better way to handle this.

Reply:
(unrelated to python3 and this pr).
Could we convert gen_protos to tox task and control its environment that way including installing grpcio and future?

Reply:
Opened https://issues.apache.org/jira/browse/BEAM-5325 to track this.